$MirOS: ports/www/lynx/patches/patch-lynx_cfg,v 1.4 2006/10/02 23:22:07 tg Exp $
--- lynx.cfg.orig	Tue Sep 19 00:28:05 2006
+++ lynx.cfg	Tue Oct 17 20:54:58 2006
@@ -92,12 +92,12 @@
 #
 # Normally we expect you will connect to a remote site, e.g., the Lynx starting
 # site:
-STARTFILE:http://lynx.isc.org/
+#STARTFILE:http://lynx.isc.org/
 #
 # As an alternative, you may want to use a local URL.  A good choice for this is
 # the user's home directory:
 .ex
-#STARTFILE:file://localhost/~/
+STARTFILE:file://localhost/~/
 #
 # Your choice of STARTFILE should reflect your site's needs, and be a URL that
 # you can connect to reliably.  Otherwise users will become confused and think
@@ -396,7 +396,10 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # Lynx (case insensitive).
 # Find RFC 1345 at http://www.ics.uci.edu/pub/ietf/uri/rfc1345.txt .
 #
-#CHARACTER_SET:iso-8859-1
+# Rationale to set this in MirBSD: NetBSD wscons assumes latin1 display
+# charsets, and lynx cannot yet check locale information to determine a
+# utf-8 environment (why?).
+CHARACTER_SET:iso-8859-1
 
 .h2 LOCALE_CHARSET
 # LOCALE_CHARSET overrides CHARACTER_SET if true, using the current locale to
@@ -420,7 +423,9 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # ASSUME_CHARSET can also be changed via the 'o'ptions menu but will
 # not be saved as permanent value in user's .lynxrc file to avoid more chaos.
 #
-#ASSUME_CHARSET:iso-8859-1
+# Rationale to set this in MirBSD: XHTML/1.0 and XHTML/1.1 are XML
+# applications, which implies use of UTF-8 Unicode by default.
+ASSUME_CHARSET:utf-8
 
 
 .h2 ASSUMED_DOC_CHARSET_CHOICE
@@ -538,7 +543,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # treatment.  There is no default, and you probably should leave this
 # undefined unless necessary.
 #
-#ASSUME_UNREC_CHARSET:iso-8859-1
+ASSUME_UNREC_CHARSET:utf-8
 
 .h2 PREFERRED_LANGUAGE
 # PREFERRED_LANGUAGE is the language in MIME notation (e.g., "en",
@@ -567,7 +572,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # the sending of an unacceptable response is also allowed.  See RFC 2068
 # (http://www.ics.uci.edu/pub/ietf/uri/rfc2068.txt).
 #
-#PREFERRED_CHARSET:
+PREFERRED_CHARSET:utf-8
 
 
 .h2 CHARSETS_DIRECTORY
@@ -648,7 +653,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # Set the threshold # of lines Lynx must render before it
 # redraws the screen in PARTIAL mode.  Anything < 0 implies
 # use of the screen size.
-#PARTIAL_THRES:-1
+PARTIAL_THRES:3
 
 
 .h2 SHOW_KB_RATE
@@ -660,7 +665,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 #	KB,ETA to show Kilobytes/second with estimated completion time.
 #	BYTES,ETA to show BYTES/second with estimated completion time.
 # Note that the "ETA" values are available if USE_READPROGRESS was defined.
-#SHOW_KB_RATE:TRUE
+SHOW_KB_RATE:BYTES,ETA
 
 .h2 SHOW_KB_NAME
 # Set the abbreviation for Kilobytes (1024).
@@ -807,8 +812,8 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # number of documents to cache (rather than the maximum number only if
 # DEFAULT_VIRTUAL_MEMORY_SIZE has been exceeded, as with VAXC/VAX).
 #
-#DEFAULT_CACHE_SIZE:10
-#DEFAULT_VIRTUAL_MEMORY_SIZE:512000
+DEFAULT_CACHE_SIZE:512
+DEFAULT_VIRTUAL_MEMORY_SIZE:25165824
 
 
 .h2 SOURCE_CACHE
@@ -824,7 +829,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # NONE is the default; the document source is not cached, and is reloaded
 #   from the network when needed.
 #
-#SOURCE_CACHE:NONE
+SOURCE_CACHE:MEMORY
 
 
 .h2 SOURCE_CACHE_FOR_ABORTED
@@ -834,7 +839,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # be preserved (and used as cache), if set to DROP lynx will drop the
 # source cache for that document (i.e. only completely downloaded documents
 # will be cached in that case).
-#SOURCE_CACHE_FOR_ABORTED:DROP
+SOURCE_CACHE_FOR_ABORTED:KEEP
 
 .h2 ALWAYS_RESUBMIT_POSTS
 # If ALWAYS_RESUBMIT_POSTS is set TRUE, Lynx always will resubmit forms
@@ -1072,7 +1077,8 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # default or that defined here can be toggled via the -force_secure
 # command line switch.
 #
-#FORCE_SSL_COOKIES_SECURE:FALSE
+# MirBSD: I wonder why this was not set to TRUE by default.
+FORCE_SSL_COOKIES_SECURE:TRUE
 
 
 .h1 Internal Behavior
@@ -1259,7 +1265,8 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # line break.  Note that the valid way to insert extra blank lines in HTML
 # is via a PRE block with only newlines in the block.
 #
-#COLLAPSE_BR_TAGS:TRUE
+# MirBSD: it's not valid, but *sigh* who bothers... damn webdesigners.
+COLLAPSE_BR_TAGS:FALSE
 
 
 .h2 TAGSOUP
@@ -1362,7 +1369,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # The default is FALSE, so that the feature needs to be enabled here
 # explicitly if you want it.
 #
-#PERSISTENT_COOKIES:FALSE
+PERSISTENT_COOKIES:TRUE
 
 
 .h2 COOKIE_FILE
@@ -1371,7 +1378,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # USE_PERSISTENT_COOKIES and the PERSISTENT_COOKIES option is enabled.
 # The cookie file can also be specified in .lynxrc or on the command line.
 #
-#COOKIE_FILE:~/.lynx_cookies
+#COOKIE_FILE:~/.etc/cookies.txt
 
 
 .h2 COOKIE_SAVE_FILE
@@ -1384,7 +1391,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # COOKIE_FILE if it is not set.  With a non-interactive Lynx session (e.g.,
 # -dump), cookies will only be saved to file if COOKIE_SAVE_FILE is set.
 #
-#COOKIE_SAVE_FILE:~/.lynx_cookies
+#COOKIE_SAVE_FILE:~/.etc/cookies.txt
 
 
 .h1 Mail-related
@@ -1525,7 +1532,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # It will be prepended by the user's home directory.
 # NOTE that a file ending in .html or other suffix mapped to text/html
 # should be used to ensure its treatment as HTML.  The built-in default
-# is lynx_bookmarks.html.  On both Unix and VMS, if a subdirectory off of
+# is ./.etc/bookmark.htm.  On both Unix and VMS, if a subdirectory off of
 # the HOME directory is desired, the path should begin with "./" (e.g.,
 # ./BM/lynx_bookmarks.html), but the subdirectory must already exist.
 # Lynx will create the bookmark file, if it does not already exist, on
@@ -1537,7 +1544,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # (see below), via the 'o'ptions menu, and can save those definitions
 # in the .lynxrc file.
 #
-#DEFAULT_BOOKMARK_FILE:lynx_bookmarks.html
+#DEFAULT_BOOKMARK_FILE:./.etc/bookmark.htm
 
 
 .h2 MULTI_BOOKMARK_SUPPORT
@@ -1916,7 +1923,8 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # prefer the more conservative action of requiring an explicit Y or y to
 # confirm.  The default defined here will override that in userdefs.h.
 #
-#QUIT_DEFAULT_YES:TRUE
+# MirBSD: pull in back default behaviour of MirBSD #2 Lynx.
+QUIT_DEFAULT_YES:FALSE
 
 
 .h1 HTML Parsing
@@ -2253,7 +2261,7 @@ MINIMAL_COMMENTS:TRUE
 # XLOADIMAGE_COMMAND definition.
 # If you use xli, you may want to add the -quiet flag.
 #
-#XLOADIMAGE_COMMAND:xli %s &
+#XLOADIMAGE_COMMAND:xloadimage %s &
 
 .h2 VIEWER
 # MIME types may be assigned to external viewers using
@@ -3204,7 +3212,9 @@ COLOR:6:brightred:black
 # otherwise. Set this setting to TRUE if you prefer numbered links, but wish
 # to get valid HTML source when printing or mailing when in psrc view.
 # Default is FALSE.
-#PRETTYSRC_VIEW_NO_ANCHOR_NUMBERING:FALSE
+#
+# MirBSD: "view source" should yield valid (more or less) HTML, at least.
+PRETTYSRC_VIEW_NO_ANCHOR_NUMBERING:TRUE
 
 
 .h1 HTML Parsing
@@ -3237,13 +3247,11 @@ COLOR:6:brightred:black
 
 .h2 JUSTIFY
 # JUSTIFY - Appearance
-# This option mirrors command-line option with same name.  Default is TRUE.  If
-# true, most of text (except headers and like this) will be justified.  This
-# has no influence on CJK text rendering.
+# This option mirrors command-line option with same name.  Default is FALSE.
+# If true, most of text (except headers and like this) will be justified.
+# This has no influence on CJK text rendering.
 #
-# This option is only available if Lynx was compiled with EXP_JUSTIFY_ELTS.
-#
-#JUSTIFY:FALSE
+JUSTIFY:TRUE
 
 .h2 JUSTIFY_MAX_VOID_PERCENT
 # JUSTIFY_MAX_VOID_PERCENT - Appearance
@@ -3271,7 +3279,8 @@ COLOR:6:brightred:black
 # explicit activation can also be requested with the -tna command line
 # option.
 #
-#TEXTFIELDS_NEED_ACTIVATION:FALSE
+# MirBSD: this is a surprise! But yes, I always wanted this.
+TEXTFIELDS_NEED_ACTIVATION:TRUE
 
 .h2 LEFTARROW_IN_TEXTFIELD_PROMPT
 # LEFTARROW_IN_TEXTFIELD_PROMPT
@@ -3333,26 +3342,26 @@ COLOR:6:brightred:black
 #ENABLE_LYNXRC:emacs_keys:ON
 #ENABLE_LYNXRC:file_editor:ON
 #ENABLE_LYNXRC:file_sorting_method:ON
-#ENABLE_LYNXRC:force_cookie_prompt:OFF
-#ENABLE_LYNXRC:force_ssl_prompt:OFF
+ENABLE_LYNXRC:force_cookie_prompt:ON
+ENABLE_LYNXRC:force_ssl_prompt:ON
 #ENABLE_LYNXRC:kblayout:ON
 #ENABLE_LYNXRC:keypad_mode:ON
 #ENABLE_LYNXRC:lineedit_mode:ON
 #ENABLE_LYNXRC:locale_charset:ON
-#ENABLE_LYNXRC:make_links_for_all_images:OFF
-#ENABLE_LYNXRC:make_pseudo_alts_for_inlines:OFF
+ENABLE_LYNXRC:make_links_for_all_images:ON
+ENABLE_LYNXRC:make_pseudo_alts_for_inlines:ON
 #ENABLE_LYNXRC:multi_bookmark:ON
 #ENABLE_LYNXRC:personal_mail_address:ON
 #ENABLE_LYNXRC:preferred_charset:ON
-#ENABLE_LYNXRC:preferred_encoding:OFF
+ENABLE_LYNXRC:preferred_encoding:ON
 #ENABLE_LYNXRC:preferred_language:ON
-#ENABLE_LYNXRC:preferred_media_types:OFF
+ENABLE_LYNXRC:preferred_media_types:ON
 #ENABLE_LYNXRC:raw_mode:OFF
 #ENABLE_LYNXRC:run_all_execution_links:ON
 #ENABLE_LYNXRC:run_execution_links_on_local_files:ON
 #ENABLE_LYNXRC:scrollbar:OFF
 #ENABLE_LYNXRC:select_popups:ON
-#ENABLE_LYNXRC:set_cookies:OFF
+ENABLE_LYNXRC:set_cookies:ON
 #ENABLE_LYNXRC:show_color:ON
 #ENABLE_LYNXRC:show_cursor:ON
 #ENABLE_LYNXRC:show_dotfiles:ON
